### PR TITLE
Align better with WMA

### PR DESCRIPTION
**Pull request by rocky**
Graph Construction & Representation:
https://reference.wolfram.com/language/guide/GraphConstructionAndRepresentation.html

Move eval stuff to pymathics.graph.eval

@mmatera when I run tests I am seeing 9 tests fail now rather than 8. I am not sure where the additional test is coming from.  I am too frazzled right now to investigate. 

Also, I notice that we now have common docpipeline code 3 places now and if we continue this direction with pymathics-natlang there will be 4 places. We should DRY this code and add the common code to a new module `mathics.doc.common_docpipeline`.  (Please let is not add to `mathics.doc.common_doc` which is large enough as it is and does related but separate stuff.

### CODE REVIEW

**Comment by mmatera** [comment]
Regarding docpipeline, I didn't copy here, but I am using the one in mathics-core. The idea is to reuse that one on each pymathics module.

File: test/test_generators.py [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-from test.helper import check_evaluation, evaluate, evaluate_value
-
-
-def setup_module(module):
-    """Load pymathics.graph"""
-    assert evaluate_value('LoadModule["pymathics.graph"]') == "pymathics.graph"
-    evaluate("SortList[list_] := Sort[Map[Sort, list]]")
-
-
-def test_connected_components():
```
Comment (mmatera) [inline test/test_generators.py:14]: Where did this test go?

**Review by mmatera — APPROVED** [review]
LGTM

File: test/test_generators.py (hunk 1 above)
Comment (rocky) [reply test/test_generators.py:14]: In master I am seeing this test twice. Once in test_algorithms.py, and once in test_generators. 

But code in  "algorithms" has been moved to components and other places. And there was a test in that which should have been in the non-existent `test.test_base`. 

Commit fbd8837 addresses both of these.

**Pull request merged**